streams/jvm/src/test/scala/zio/stream/StreamSpec.scala

```diff
@@ -707,6 +710,12 @@ class ZStreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
       slurp(leftStream) must_=== slurp(rightStream)
     }
 
+  private def flatMapParGuaranteeOrdering = prop { m: List[Int] =>
```

Comment:
As I mentioned - super important test to have! Great thinking here.

Reply:
Thanks. Couldn't have done it without your help.